### ISSUE TITLE
502 - Bad Gateway

### ISSUE THREAD

**Issue by CyberMixRo** [AzuraCast/ansible-installer#1]
Hello,

I've try to install Ansible version using your installer on 2 fresh Ubuntu server versions: 

- Ubuntu 22.04 "Jammy" LTS (Recommended)
- Ubuntu 20.04 "Focal" LTS

On both cases i've got same results, 502 Bad Gateway (nginx). Looking on Supervisor also sftpgo do not start. I've try also old ansible azuracast installer, same results.

I will appreciate if someone have a solution to can install ansible version, i'm interested to customize and add some functions, i can do it over docker but take to much to unpack and pack back.

Thanks

**Comment by gunsar** [comment]
Yes, I tried fresh install VPS and experienced the same 502 Bad Gateway problem on both Ubuntu 20 and 22. It looks like a new setting has to be changed for the latest version of Nginx

**Comment by CyberMixRo** [comment]
Many get errors on Ansible version because of Supervisor configs, in my case MariaDB and PHP-FPM services are not startion, maybe that help

**Comment by isAAAc** [comment]
`systemctl start php8.2-fpm.service` and then i can "first register" on the webui

**Comment by gunsar** [comment]
@isAAAc 

I tried again fresh install. install stops here (see picture), opens in browser 502 bad gateway, uses, then enters systemctl start php8.2-fpm.service command, directly vps can't open anything

![Screenshot 2023-02-20 130102](https://user-images.githubusercontent.com/56213562/220022988-542beeeb-6c6e-46df-9ac2-08fa6125797f.jpg)

**Comment by isAAAc** [comment]
yes here it continues: 
```
[...]
TASK [azuracast-build : Install NodeJS and NPM] ************************************************************************************
changed: [localhost]                                

TASK [azuracast-build : Set permissions for folders] *******************************************************************************
changed: [localhost] => (item=/var/azuracast/www/frontend/node_modules)                                  
changed: [localhost] => (item=/var/azuracast/www/web/static)                                             
changed: [localhost] => (item=/var/azuracast/www/web/static/dist)                                        
changed: [localhost] => (item=/var/azuracast/www/web/static/webpack_dist)                                

TASK [azuracast-build : Pull Node Dependencies] ************************************************************************************
changed: [localhost]                                

TASK [azuracast-build : Build AzuraCast Frontend Scripts] **************************************************************************
changed: [localhost]                                

TASK [azuracast-build : Set permissions of generated frontend content] *************************************************************
changed: [localhost] => (item=chown -R azuracast:www-data /var/azuracast/www)                            
changed: [localhost] => (item=find /var/azuracast/www -type d -exec chmod 755 {} \;)
changed: [localhost] => (item=find /var/azuracast/www -type f -exec chmod 644 {} \;)

TASK [azuracast-setup : Ensure update script continues to be executable] ***********************************************************
changed: [localhost] => (item=/var/azuracast/ansible/update.sh)                                          
changed: [localhost] => (item=/var/azuracast/www/bin/console)                                            

TASK [azuracast-setup : Run AzuraCast Setup (Install Mode)] ************************************************************************
changed: [localhost] 
[...]
```
strange

**Comment by isAAAc** [comment]
after a reboot, php and mariadb need to be manually started via systemctl.

supervisor don't want to start services
```
SupervisorException at /var/azuracast/www/src/Exception/SupervisorException.php L69: php-worker encountered an error: SpawnErrorException
```

**Comment by gunsar** [comment]
I tried to reinstall VPS and Azuracast Ansible again. the final result :



TASK [azuracast-build : Pull Node Dependencies] ********************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["npm", "ci"], "delta":              "0:06:59.666588", "end": "2023-02-20 18:22:31.472112", "msg": "non-zero return c             ode", "rc": 1, "start": "2023-02-20 18:15:31.805524", "stderr": "npm ERR! code E             HOSTUNREACH\nnpm ERR! syscall connect\nnpm ERR! errno EHOSTUNREACH\nnpm ERR! req             uest to https://registry.npmjs.org/zxcvbn/-/zxcvbn-4.4.2.tgz failed, reason: con             nect EHOSTUNREACH 2606:4700::6810:1823:443\n\nnpm ERR! A complete log of this ru             n can be found in:\nnpm ERR!     /var/azuracast/.npm/_logs/2023-02-20T18_15_32_3             45Z-debug-0.log", "stderr_lines": ["npm ERR! code EHOSTUNREACH", "npm ERR! sysca             ll connect", "npm ERR! errno EHOSTUNREACH", "npm ERR! request to https://registr             y.npmjs.org/zxcvbn/-/zxcvbn-4.4.2.tgz failed, reason: connect EHOSTUNREACH 2606:             4700::6810:1823:443", "", "npm ERR! A complete log of this run can be found in:"             , "npm ERR!     /var/azuracast/.npm/_logs/2023-02-20T18_15_32_345Z-debug-0.log"]             , "stdout": "", "stdout_lines": []}

RUNNING HANDLER [azuracast-user : send user notification] **********************

RUNNING HANDLER [azuracast-db-install : send mysql notification] ***************

PLAY RECAP *********************************************************************
localhost                  : ok=99   changed=82   unreachable=0    failed=1    s             kipped=6    rescued=0    ignored=2



the install didn't work and couldn't do anything

**Comment by isAAAc** [comment]
there is no way to keep on the stable working ansible deploy version ? o0

**Comment by isAAAc** [comment]
@BusterNeece , do you encounter the same problems ?

**Comment by ItsMitchh** [comment]
Have patience, Ansible isn't supported by us, as stated in the readme. Community contributions are welcome.

**Comment by isAAAc** [comment]
thx @gunsar <3
work in progress, but install is KO too by this way for now (i opened an issue too)
